src/rust/engine/src/nodes.rs

```diff
@@ -457,23 +412,23 @@ impl Step for SelectProjection {
         self.variants.clone()
       );
     let dep_product =
-      match context.get(&input_node) {
-        Some(&Complete::Return(ref value)) =>
-          value,
-        Some(&Complete::Noop(_, _)) =>
+      match context.graph.get(&input_node) {
+        &Some(Complete::Return(ref value)) =>
+          context.externs.clone_val(value),
```

Comment:
thank you for explaining this earlier - was confused by why you couldnt just wrap the reference to value here and other similar occurrences. returning a ref would mean that you continue to borrow mutably here (which you dont need to here).